build-tools/manifest-attribute-codegen/README.md

```diff
@@ -1,11 +1,105 @@
 # What is this?
 
-manifest-attribute-codegen.exe is a simple tool that collects Android SDK
-resource information about AndroidManifest.xml, then generate a unified
+`manifest-attribute-codegen` a tool that collects Android SDK
+resource information about what XML elements can be placed in an AndroidManifest.xml, then generate a unified
 element/attribute definition with "which API introduced this" information.
 
-With that, we can find what needs to be added to xamarin-android.
+Using this information, we can generate `[(Application|Activity|etc)Attribute]` classes that can be used
+by both `Mono.Android` and `Xamarin.Android.Build.Tasks`.
+
+This generally only needs to be done each time we bind a new Android API level.
+
+# How to use
+
+Ensure all Android platform SDK levels are installed.  This can be done with `xaprepare`:
+
+```
+xaprepare android-sdk-platforms=all
+```
+
+From this directory, run:
+
+```
+dotnet build -t:GenerateManifestAttributes
+```
+
+If all found elements/attributes are accounted for in `metadata.xml`, new `*Attribute.cs` files
```

Comment:
I'd like some docs for the structure & semantics of `metadata.xml`.  It has `<type/>` and `<element/>` attributes; what do they *mean*?  How do they relate?  What are the semantics?

For example:

```xml
<type
    name="activity"
    namespace="Android.App"
    outputFile="src\Xamarin.Android.NamingCustomAttributes\Android.App\ActivityAttribute.cs"
    usage="AttributeTargets.Class" jniNameProvider="true"
/>
```

Where does `activity` come from?

Or for `<element/>`:

```xml
 <element path="activity.exported" type="bool" />
```

Where does `activity.exported` come from? 

Is there any relation between these values and `data/res/values/attrs_manifest.xml`?